bitnami/mongodb/templates/replicaset/statefulset.yaml

```diff
@@ -372,14 +376,29 @@ spec:
                 name: {{ include "mongodb.secretName" . }}
                 key: mongodb-root-password
           {{- end }}
+          volumeMounts:
+          {{- if .Values.tls.enabled }}
+          - name: certs
+            mountPath: /certs
+          {{- end }}
           ports:
             - name: metrics
               containerPort: 9216
           {{- if .Values.metrics.livenessProbe.enabled }}
           livenessProbe:
-            httpGet:
-              path: /metrics
-              port: metrics
+            exec:
+              command:
+                - sh
+                - -c
+              {{- if .Values.tls.enabled }}
+                - >-
+                  /bin/mongodb_exporter --mongodb.uri "mongodb://root:$(echo $MONGODB_ROOT_PASSWORD | sed -r "s/@/%40/g;s/:/%3A/g")@localhost:27017/admin?tls=true&tlsCertificateKeyFile=/certs/mongodb.pem&tlsCAFile=/certs/mongodb-ca-cert{{ .Values.metrics.extraUri }}" {{ .Values.metrics.extraFlags }}
+                  --test
+              {{- else }}
+                - >-
+                  /bin/mongodb_exporter --mongodb.uri "mongodb://root:$(echo $MONGODB_ROOT_PASSWORD | sed -r "s/@/%40/g;s/:/%3A/g")@localhost:27017/admin{{ .Values.metrics.extraUri }}" {{ .Values.metrics.extraFlags }}
+                  --test
+              {{- end }}
```

Comment:
Maybe I'm wrong but this command is testing if the `/bin/mongodb_exporter` configuration passed to the command is ok, isn't it?

Instead we should keep the previous liveness and readiness probe, right?

Reply:
Yes you are correct in saying that command is testing if the /bin/mongodb_exporter config passed to the command is ok or not. 
However the way the previous liveness/readiness probe is setup, its impossible for it to do a check if TLS is enabled because there is no way or format to add TLS settings or anything else to that command. 
```
livenessProbe:
            httpGet:
              path: /metrics
              port: metrics
```
Basically with the previous setup, to perform a probe, the kubelet sends an HTTP GET request to the server that is running in the container and listening on port 9216. If the handler for the server's /metrics path returns a success code, the kubelet considers the container to be alive and healthy. If the handler returns a failure code, the kubelet kills the container and restarts it. Any code greater than or equal to 200 and less than 400 indicates success. Any other code indicates failure.
There is no way (at least to my knowledge) to specify additional strings or TLS parameters for this probe's command

However I did see in a previous chart, they used the format I proposed in which the /bin/mongodb_exporter takes in the mongodb.uri string with --test appended to it to check the mongodb connection status, print the buildInfo() information and then exit. This to me is equivalent to the previous way but also allows us to specify a connection string which can take in our TLS settings for a TLS enabled cluster.